### PR TITLE
added undistorted output and used image transport

### PR DESCRIPTION
**Pull request by ZacharyTaylor**
Added in an output giving the undistorted camera info
Changed to image transport

### CODE REVIEW

**Comment by ZacharyTaylor** [comment]
So I ended up pretty much rewriting everything. It now
- uses image transport
- can take in either a kalibr yaml file or a ros topic for the input camera info
- can generate an output that emulates any camera via a loaded yaml file giving the output parameters (if not given will just copy the input cameras)
- can just add a cam_info topic to an image topic without any other processing
- uses fixed point opencv maps and eigen maps to the variables making it a lot more efficient.

File: cfg/example_input_config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+example_input_camera:
```
Comment (helenol) [inline cfg/example_input_config.yaml]: Awesome, thanks for adding these :)

File: src/image_undistort.cpp [hunk 2]
```diff
@@ -3,59 +3,282 @@
...
+      camera_info_in_.P.data());
+
+  Eigen::Map<const Eigen::Matrix<double, 3, 4, Eigen::RowMajor>> P_out(
+      camera_info_out_.P.data());
+
+  cv::Size resolution(camera_info_out_.width, camera_info_out_.height);
 
   bool using_radtan;
-  if (camera_msg->distortion_model == std::string("radtan")) {
+  if ((camera_info.distortion_model == std::string("Plumb Bob")) ||
```
Comment (helenol) [inline src/image_undistort.cpp]: Maybe convert to lowercase so it's case independent (I've seen "plumb bob" more often than "Plumb Bob")?

http://math-linux.com/c/faq-c/faq-c-stl/article/how-to-convert-string-to-lower-case-or-upper-case-in-c is one example.

File: src/image_undistort.cpp [hunk 3]
```diff
@@ -3,59 +3,282 @@
...
   }
 
-  double zoom;
-  private_nh_.param("zoom", zoom, kDefaultZoom);
+  undistorter_ptr_ = std::make_shared<Undistorter>(
+      resolution, P_in, P_out, using_radtan, camera_info_in_.D);
+}
+
+bool ImageUndistort::loadCameraParameters(
+    sensor_msgs::CameraInfo* loaded_camera_info, std::string* image_topic,
```
Comment (helenol) [inline src/image_undistort.cpp]: Parameter order: is_input should be first, in/out variables last.

**Pull request merged**